chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -260,7 +260,15 @@ post_state() {
 # -----------------------
 
 MAC=$(cat /sys/class/net/$BOOTDEV/address)
+<% if node[:provisioner][:keep_existing_hostname] -%>
+HOSTNAME=$(cat /etc/HOSTNAME)
+if ! echo "$HOSTNAME" | grep -q -o "$DOMAIN$"; then
+  echo "The fully qualified domain name did not contain the cloud domain."
+  exit 1
+fi
+<% else -%>
 HOSTNAME="d${MAC//:/-}.${DOMAIN}"
+<% end -%>
 sed -i -e "s/\(127\.0\.0\.1.*\)/127.0.0.1 $HOSTNAME ${HOSTNAME%%.*} localhost.localdomain localhost/" /etc/hosts
 echo "$HOSTNAME" > /etc/HOSTNAME
 hostname "$HOSTNAME"
```

Comment:
Normally we can skip this lines but in this case they didn't cause any unwanted changes.